src/main/java/com/massivecraft/factions/data/MemoryBoard.java

```diff
@@ -272,81 +272,84 @@ public int getFactionCoordCountInWorld(Faction faction, String worldName) {
      * The map is relative to a coord and a faction north is in the direction of decreasing x east is in the direction
      * of decreasing z
      */
-    public ArrayList<FancyMessage> getMap(FPlayer fplayer, FLocation flocation, double inDegrees) {
```

Comment:
This contains so many renames I can't tell what's actually changed beyond seeing some brackets removed (that I'd just add back in anyway).

Reply:
Take a looksie and compare readability with [previous](https://github.com/drtshock/Factions/blob/1.6.x/src/main/java/com/massivecraft/factions/data/MemoryBoard.java#L280).

```suggestion
    public ArrayList<FancyMessage> getMap(FPlayer fPlayer, FLocation fLocation, double inDegrees) {
        final MainConfig conf = FactionsPlugin.getInstance().conf();
        final ArrayList<FancyMessage> out = new ArrayList<>();

        final Faction senderFaction = fPlayer.getFaction();
        final Faction senderFactionAtLocation = getFactionAt(fLocation);

        out.add(new FancyMessage(FactionsPlugin.getInstance()
            .txt().titleize("(" + fLocation.getCoordString() + ") " + senderFactionAtLocation.getTag(fPlayer))));

        final ArrayList<String> compass =
            AsciiCompass.getAsciiCompass(inDegrees, ChatColor.RED, FactionsPlugin.getInstance().txt().parse("<a>"));

        final int halfMapWidth = conf.map().getWidth() / 2;
        final int halfMapHeight = fPlayer.getMapHeight() / 2;
        final int mapWidth = halfMapWidth * 2 + 1;
        final int mapHeight = conf.map().isShowFactionKey() ? halfMapHeight * 2 : halfMapHeight * 2 + 1;

        final FLocation topLeft = fLocation.getRelative(-halfMapWidth, -halfMapHeight);
        final Map<String, Character> factionsList = new HashMap<>();

        int charIndex = 0;

        // ... for each row
        for (int dz = 0; dz < mapHeight; dz++) {
            final FancyMessage row = new FancyMessage("");

            // ... add ascii compass
            if (dz < 3) row.then(compass.get(dz));

            for (int dx = (dz < 3 ? 6 : 3); dx < mapWidth; dx++) {
                // ... current chunk fplayer is standing in
                if (dx == halfMapWidth && dz == halfMapHeight) {
                    row.then("+").color(ChatColor.AQUA).tooltip(TL.CLAIM_YOUAREHERE.toString());
                    continue;
                }

                final Faction factionAt = getFactionAt(topLeft.getRelative(dx, dz));
                final Relation relation = fPlayer.getRelationTo(factionAt);

                if (factionAt.isWilderness()) {
                    row.then("-").color(conf.colors().factions().getWilderness());
                } else if (factionAt.isSafeZone()) {
                    row.then("+").color(conf.colors().factions().getSafezone());
                } else if (factionAt.isWarZone()) {
                    row.then("+").color(conf.colors().factions().getWarzone());
                }

                // ... check relations
                else if (factionAt == senderFaction || factionAt == senderFactionAtLocation || relation.isAtLeast(Relation.ALLY) ||
                    conf.map().isShowNeutralFactions() && relation.equals(Relation.NEUTRAL) ||
                    conf.map().isShowEnemyFactions() && relation.equals(Relation.ENEMY) ||
                    conf.map().isShowTruceFactions() && relation.equals(Relation.TRUCE)) {

                    if (!factionsList.containsKey(factionAt.getTag())) {
                        factionsList.put(factionAt.getTag(), this.mapKeyChrs[Math.min(charIndex++, this.mapKeyChrs.length - 1)]);
                    }

                    final char tag = factionsList.get(factionAt.getTag());
                    row.then(String.valueOf(tag)).color(factionAt.getColorTo(senderFaction));
                } else {
                    row.then("-").color(ChatColor.GRAY);
                }
            }
            out.add(row);
        }

        // ... add each factions' name with key
        if (FactionsPlugin.getInstance().conf().map().isShowFactionKey()) {
            final FancyMessage row = new FancyMessage("");
            for (String key : factionsList.keySet()) {
                final Relation relation = fPlayer.getRelationTo(Factions.getInstance().getByTag(key));
                row.then(String.format("%s: %s ", factionsList.get(key), key)).color(relation.getColor());
            }
            out.add(row);
        }

        return out;
    }
```